src/main.c

```diff
@@ -33,12 +33,43 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
 //#define VERSION				"0.1" //gets set via autotools
 #define GRID_LINES				19
 #define GRID_COLS				37
+
+#ifndef SMALL_DISPLAY
```

Comment:
please create one block #ifdef SMAPP_DISPLAY and do all the things in there, and one #else block where you do the rest. for the defines.